maplocate/admin/users.py

```diff
@@ -37,6 +111,61 @@ def user_update(self):
     def user_delete(self):
         pass
 
+    @validate(FilterUserForm, check_param='json_body')
     @asyncio.coroutine
-    def users_list(self):
-        pass
+    def users_list(self, request, form):
+        """List users.
+        Request: 'GET', 'admin/user/'
+        """
+
+        yield from self.auth_admin_session(request, Permission.user_view)
+        with (yield from self.postgres) as pg_con:
+            query = db.user.select()
+            if form['filter']['fullname']:
+                fname, *lname_tail = form['filter']['fullname'].split(' ')
+                if lname_tail:
+                    expr = or_(db.user.c.firstname.like(fname + '%'),
+                               *[db.user.c.lastname.like(word + '%')
+                                 for word in lname_tail])
+                else:
+                    expr = or_(db.user.c.firstname.like(fname + '%'),
+                               db.user.c.lastname.like(fname + '%'))
+                query = query.where(expr)
+
+            if form['filter']['email']:
+                query = query.where(
+                    db.user.c.login.like(form['filter']['email'] + '%')
+                )
+
+            cursor = yield from pg_con.execute(
+                query.order_by('firstname', 'lastname')
+            )
```

Comment:
Good way to get list of dicts from RowProxy
users = yield from cursor.fetchall()
users = list(map(dict, users))